src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -508,40 +575,71 @@ class DetailsCard extends React.Component {
       vm.get('canUserEditVm') &&
       vm.getIn(['pool', 'id']) === undefined
 
+    // Status
     const status = vm.get('status')
     const uptime = formatUptimeDuration({ start: vm.get('startTime') })
 
+    // Host Name
     const hostName = hosts && hosts.getIn([vm.get('hostId'), 'name'])
 
+    // IP Addresses
     const ip4Addresses = !vm.has('nics') ? [] : vm.get('nics').reduce((ipSet, nic) => [...ipSet, ...nic.get('ipv4')], [])
     const ip6Addresses = !vm.has('nics') ? [] : vm.get('nics').reduce((ipSet, nic) => [...ipSet, ...nic.get('ipv6')], [])
 
+    // FQDN
     const fqdn = vm.get('fqdn')
 
+    // Cluster
     const canChangeCluster = vmCanChangeCluster(status)
     const clusterId = vm.getIn(['cluster', 'id'])
-    const clusterName = (clusters && clusters.getIn([clusterId, 'name'])) || msg.notAvailable()
+    const cluster = clusters && clusters.get(clusterId)
+    const clusterName = (cluster && cluster.get('name')) || msg.notAvailable()
 
+    // Data Center
     const dataCenterId = (clusters && clusters.getIn([clusterId, 'dataCenterId']))
     const dataCenter = dataCenters && dataCenters.find(v => v.id === dataCenterId)
     const dataCenterName = (dataCenter && dataCenter.name) || msg.notAvailable()
 
+    // Template
     const templateId = vm.getIn(['template', 'id'])
     const templateName = (templates && templates.getIn([templateId, 'name'])) || msg.notAvailable()
 
+    // CD
     const canChangeCd = vmCanChangeCd(status)
     const cdImageId = vm.getIn(['cdrom', 'fileId'])
     const cdImage = isoList.find(iso => iso.file.id === cdImageId)
     const cdImageName = (cdImage && cdImage.file.name) || `[${msg.empty()}]`
 
-    const couldInitEnabled = vm.getIn(['cloudInit', 'enabled'])
+    // Cloud-Init
+    const cloudInitEnabled = vm.getIn(['cloudInit', 'enabled'])
+    const cloudInitHostName = vm.getIn(['cloudInit', 'hostName'])
+    const cloudInitSshAuthorizedKeys = vm.getIn(['cloudInit', 'sshAuthorizedKeys'])
 
+    // Boot Menu
     const bootMenuEnabled = vm.get('bootMenuEnabled')
 
+    // Optimized for
     const optimizedFor = rephraseVmType(vm.get('type'))
 
+    // VCPU
     const vCpuCount = vm.getIn(['cpu', 'vCPUs'])
 
+    // Boot devices
+    const allowedBootDevices = ['hd', 'network', 'cdrom']
+    const FIRST_DEVICE = 0
+    const SECOND_DEVICE = 1
+    const bootDevices = []
+
+    for (let i = 0; i < defaultBootDevices.size; i++) {
```

Comment:
Why even bother with an array if we know we only support exactly 2 devices? For example:
````js
    const bootDevices = [
      vm.getIn(['os', 'bootDevices', 0], defaultBootDevices.get(0)),
      vm.getIn(['os', 'bootDevices', 1], defaultBootDevices.get(1)),
    ]
````